src/PaymentGateways/Gateways/Stripe/Actions/GetStripeGatewayData.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe\Actions;
+
+use Give\Donations\Models\Donation;
+use Give\Framework\Exceptions\Primitives\Exception;
+use Give\Framework\PaymentGateways\Log\PaymentGatewayLog;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\PaymentMethod;
+use Stripe\Exception\ApiErrorException;
+
+class GetStripeGatewayData
+{
+    /**
+     * Returns $gatewayData array to be used in stripe gateways.
+     *
+     * @unreleased add try / catch with exception logging
+     */
+    public function __invoke($gatewayData, Donation $donation): array
+    {
+        try {
+            $paymentMethod = (new GetPaymentMethodFromRequest())($donation);
+            /**
+             * This supplies the actual Stripe Api PaymentMethod
+             */
+            $gatewayData['paymentMethod'] = $paymentMethod;
+            /**
+             * This internal ValueObject is currently being used throughout the Stripe gateways.
+             * Eventually this array key will be @deprecated in favor of using Stripe's PaymentMethod
+             * Object directly via $gatewayData['paymentMethod']
+             */
+            $gatewayData['stripePaymentMethod'] = new PaymentMethod($paymentMethod->id);
```

Comment:
Should we go ahead and mark `stripePaymentMethod` as deprecated?

Reply:
Is it possible to mark an array item as deprecated?  I wasn't sure about that.

Reply:
I was assuming that you could and questioning the statement "Eventually this array key will be @deprecated".

That said, it doesn't look like PHPStan supports the `@deprecated` tag for array key items - unfortunately.

Reply:
The (perhaps harebrained) idea that comes to mind is that we could replace the array with an `ArrayAccess` object and then throw deprecation notices in a magic `__get` method.

Probably overkill in this situation.